[Jun Wei] iP

junwei-tan opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2223S1/ip/pull/407).
2. double-click it
3. add your tasks.
4. let it manage your tasks for you :wink:

And its **FREE!**

Features:
- [X] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

> System.out.println("Duke");

```java
System.out.println("Duke");
```

Commit:
Add Gradle support

Commit:
Duke Level-1: Greet, Echo, Exit

Commit:
Duke Level-2: Add, List

Commit:
Duke Level-3: Mark as Done

Commit:
Duke Level-4: ToDo, Event, Deadline

Commit:
Duke A-TextUiTesting: Automated Text UI Testing

Commit:
Duke Level-5: Handle Errors

Commit:
Duke Level-6: Delete

Commit:
Add Level-7: Save

Commit:
Add Level-8: Dates

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
Refactor code

Commit:
Organize into packages

Commit:
Merge remote-tracking branch 'GITHUB_add_graddle_support_branch/add-gradle-support'

gabyang commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,14 @@
+import duke.common.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
```
Just wanted to say good work on ordering the imports! It is something that people normally make mistakes on!

gabyang commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,101 @@
+        String[] commandArgs = input.trim().split(" ", 2);
+        String commandType = commandArgs[0];
+        Command command;
+        switch (commandType) {
```
I like your use of switch flow control! Much better than using a whole bunch of if-else statements!

gabyang commented on src/main/java/duke/common/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+package duke.common;
```
Don't you think it is a good idea to have more exceptions under DukeExceptions to help with different types of exceptions? I think there can be more classes inheriting from this class!

junwei-tan replied:
Thank you for the feedback.

gabyang reviewed (commented):
Overall a very readable application! I had a great time reviewing your PR and I must say it is structured better than other applications!

Commit:
Add A-Gradle

Commit:
Add JUnit tests

Commit:
Add JavaDoc

Commit:
Refactor code to comply with coding standard

Commit:
Add find feature

Commit:
Merge remote-tracking branch 'origin/branch-A-JavaDoc'

Commit:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

Commit:
Merge remote-tracking branch 'origin/branch-Level-9'

nopehax commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,230 @@
+package duke.parser;
```
I like that you have a number of tests to ensure that your methods are working correctly!

junwei-tan replied:
Thank you.

nopehax commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,230 @@
+                    exception.getMessage());
+        }
+    }
+
```
Is there a reason why you didn't add JavaDocs for the test cases? It can help explain what each test case is testing.

junwei-tan replied:
Thank you for the feedback.

nopehax reviewed (commented):
Overall great job!! I like that your JavaDocs are succinct

SeekSaveServe commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,20 @@
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+public class FindCommand extends Command {
```
Should there be a description for the initializer?

SeekSaveServe commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,20 @@
+public class FindCommand extends Command {
+    private String searchTerm;
+
+    public FindCommand(String searchTerm) {
```
Should there be a description for this function?

SeekSaveServe reviewed (approved):
Overall, this is a great commit!

Commit:
Add GUI

Commit:
Merge remote-tracking branch 'origin/branch-Level-10'

Commit:
Add CheckStyle

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Fix code style violations

Commit:
Add assertions

There are no assertions used.

This might make the code less readable as there are assumptions that
should hold in various parts of the code.

Let's add assertions to the following classes:
- AddCommand
- MarkCommand
- UnmarkCommand
- Parser
- Task
- TaskList

junwei-tan committed:
Merge pull request #1 from junwei-tan/branch-A-Assertions

Add assertions

Commit:
Improve code quality

Commit:
Merge branch 'master' into branch-A-CodeQuality

junwei-tan committed:
Merge pull request #3 from junwei-tan/branch-A-CodeQuality

branch-A-CodeQuality

Commit:
Update delete command with additional feature

Commit:
Merge branch 'branch-C-MassOps'

Commit:
Refine GUI

GUI experiences a bug where previous responses from Duke gets cropped
with ellipses.

This dimishes the user experience greatly.

Let's fix this bug and make minor changes to the fonts.

Commit:
Add Ui.png to docs folder

Commit:
Add user guide

Commit:
Add credit for reused code

Commit:
Refine user guide

The user guide does not render properly on github pages and the link to
download the program is wrong.

Poorly rendered user guides are not user-friendly.

Let's,
* Add config file to specify gfmd rendering
* Update link to download Duke
